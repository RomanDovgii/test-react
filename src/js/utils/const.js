export const MAXIMUM_USERS_PER_PAGE = 50;

export const AppRoute = {
  SELECT: `/select`,
  TABLE: `/table`
};
