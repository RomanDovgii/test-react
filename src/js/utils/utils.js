export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const sortToTop = (a, b) => {
  switch (true) {
    case a > b:
      return -1;
    case a < b:
      return 1;
    default:
      return 0;
  }
};

export const sortToBot = (a, b) => {
  switch (true) {
    case a < b:
      return -1;
    case a > b:
      return 1;
    default:
      return 0;
  }
};
