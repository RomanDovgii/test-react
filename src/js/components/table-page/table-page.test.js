import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import TablePage from "./table-page";
import {noop} from "../../utils/utils";

it(
    `Should render TablePage`,
    () => {
      const renderer = new ShallowRenderer();

      const tree = renderer
      .render(
          <TablePage
            onFilterClick={noop}
          />
      );

      expect(tree).toMatchSnapshot();
    }

);
