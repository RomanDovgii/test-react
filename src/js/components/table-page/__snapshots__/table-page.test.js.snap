// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render TablePage 1`] = `
<React.Fragment>
  <Controls />
  <withRouter(Connect(Table)) />
  <Memo(Connect(UserInfo)) />
</React.Fragment>
`;
